umm-lib/src/cmr/umm/iso_smap/granule/orbit.clj

```diff
@@ -112,4 +168,39 @@
        (when end-direction
          (let [ed (subs orbit-str
                         end-direction)]
-           (convert-direction (str/trim (subs ed (inc (.indexOf ed ":")))))))))))
+           (convert-direction (str/trim (subs ed (inc (.indexOf ed ":"))))))))
+      "OrbitCalculatedSpatialDomains"
+      (g/->OrbitCalculatedSpatialDomain
+        (when orbital-model-name
+         (let [o-m-name (subs orbit-str
+                          orbital-model-name
+                          (or orbit-number start-orbit-number stop-orbit-number
+                              equator-crossing-longitude equator-crossing-date-time (count orbit-str)))]
+           (str/trim (subs o-m-name (inc (.indexOf o-m-name ":"))))))
+        (when orbit-number
```

Comment:
I think we'll be generating invalid OrbitCalculatedSpatialDomain objects if there's any kind of unexpected value in the ISO SMAP XML. Both Cayvon and I have seen this when testing. We need to add some more validations and an error test or two to make sure that XML we can't parse correctly gets rejected on ingest with a good error message instead of ending up with an internal server error.